calico-vpp-agent/cmd/calico_vpp_dataplane.go

```diff
@@ -93,10 +96,47 @@ func main() {
 	}
 	// Once we have the api connection, we know vpp & vpp-manager are running and the
 	// state is accurately reported. Wait for vpp-manager to finish the config.
-	err = common.WaitForVppManager()
+	vppManagerInfo, err := common.WaitForVppManager()
 	if err != nil {
 		log.Fatalf("Vpp Manager not started: %v", err)
 	}
+	routeWatcher := &watchers.RouteWatcher{}
+	poolWatcher := &watchers.PoolWatcher{
```

Comment:
This is probably worth a `NewPoolWatcher(...)` and `NewRouteWatcher()` to hide the logic.